src/__tests__/utils.test.ts

```diff
@@ -379,19 +386,90 @@ describe('utils', () => {
   });
 
   describe('getIds', () => {
-    for (const input of [
-      ['123456789012345678900001', '123456789012345678900002'],
-      [new ObjectId('123456789012345678900001'), new ObjectId('123456789012345678900002')],
-      ['123456789012345678900001', new ObjectId('123456789012345678900002')],
-    ]) {
-      test('case', () => {
-        const result = getIds(input);
-
-        equal(result.length, 2);
-        ok(result[0] instanceof ObjectId);
-        equal(result[0].toHexString(), '123456789012345678900001');
-        ok(result[1] instanceof ObjectId);
-        equal(result[1].toHexString(), '123456789012345678900002');
+    const testCases: readonly [
+      string,
+      {
+        readonly input: readonly ObjectIdConstructorParameter[];
+        readonly expected: readonly ObjectId[];
+      },
+    ][] = [
+      [
+        'strings',
+        {
+          input: ['123456789012345678900001', '123456789012345678900002'],
+          expected: [
+            new ObjectId('123456789012345678900001'),
+            new ObjectId('123456789012345678900002'),
+          ],
+        },
+      ],
+      [
+        'ObjectIds',
+        {
+          input: [
+            new ObjectId('123456789012345678900099'),
+            new ObjectId('123456789012345678900022'),
+          ],
+          expected: [
+            new ObjectId('123456789012345678900099'),
+            new ObjectId('123456789012345678900022'),
+          ],
+        },
+      ],
+      [
+        'Uint8Arrays',
+        {
+          input: [
+            new Uint8Array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]),
+            new Uint8Array([13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24]),
+          ],
+          expected: [
+            new ObjectId('0102030405060708090a0b0c'),
+            new ObjectId('0d0e0f101112131415161718'),
+          ],
+        },
+      ],
+      [
+        'mixed',
+        {
+          input: ['123456789012345678900014', new ObjectId('123456789012345678900088')],
+          expected: [
+            new ObjectId('123456789012345678900014'),
+            new ObjectId('123456789012345678900088'),
+          ],
+        },
+      ],
+      [
+        'invalid values',
+        {
+          input: ['123', '123456789012345678900021'],
+          expected: [new ObjectId('123456789012345678900021')],
+        },
+      ],
+    ];
+
+    for (const testCase of testCases) {
+      const [caseName, { input, expected }] = testCase;
+
+      test(`case ${caseName}`, (t: TestContext) => {
+        t.plan(4);
+        // Given
+
+        t.assert.ok(expected.length <= input.length);
```

Comment:
How about using the imports from `node:assert/strict` that we already do in this file instead of using `t.assert.`?

Reply:
We can do that, but I think we'd actually get more bang out of using the `TestContext` more generally.

That way you don't have to pick and choose which assertions to import; instead every test file and rely on the same basic import:

```ts
import { describe, test, TestContext } from 'node:test';
```

This not only reduces import boilerplate, but also means that if a test needs to grow async behavior we can add a `.plan` call without having to modify the rest of the case.